pkg/kubectl/env_file.go

```diff
@@ -39,38 +80,23 @@ func addFromEnvFile(filePath string, addTo func(key, value string) error) error
 
 	scanner := bufio.NewScanner(f)
 	currentLine := 0
-	utf8bom := []byte{0xEF, 0xBB, 0xBF}
 	for scanner.Scan() {
+		// Proccess the current line, retrieving a key/value pair if
+		// possible.
 		scannedBytes := scanner.Bytes()
-		if !utf8.Valid(scannedBytes) {
-			return fmt.Errorf("env file %s contains invalid utf8 bytes at line %d: %v", filePath, currentLine+1, scannedBytes)
+		key, value, err := proccessEnvFileLine(scannedBytes, filePath, currentLine)
+		if err != nil {
+			return err
 		}
-		// We trim UTF8 BOM
-		if currentLine == 0 {
-			scannedBytes = bytes.TrimPrefix(scannedBytes, utf8bom)
-		}
-		// trim the line from all leading whitespace first
-		line := strings.TrimLeftFunc(string(scannedBytes), unicode.IsSpace)
 		currentLine++
```

Comment:
It makes me sad that we have to keep track of how many lines we've seen, and pass that variable to `processEnvFileLine` while it's only being used once.

Should we write a `TrimBOM` function that would take the file, read the first three bytes, and `Seek` back to the beginning if it doesn't match the bom?